src/cli/build.cpp

```diff
@@ -0,0 +1,136 @@
+    build_cli->add_option("recipe", uenv_recipe_path, "Path to uenv recipe")
+        ->required(true);
+    build_cli->footer(build_footer);
+    build_cli->callback([&settings] { settings.mode = settings.build; });
+}
+
+std::string build_footer() {
+    using enum help::block::admonition;
+    using help::lst;
+    std::vector<help::item> items{
+        // clang-format off
+        help::block{none, "Build uenv images." },
+        help::linebreak{},
+        help::block{none, fmt::format("For more information on how to use individual commands use the {} flag.", lst("--help")) },
+        help::linebreak{},
+        help::block{xmpl, "Build QuantumESPRESSO UENV" },
+        help::block{code, "uenv build --name quantumespresso --version 7.3.1 -u gh200 -s daint /path/to/QE/recipe"},
+        help::linebreak{},
+        // clang-format on
+    };
+
+    return fmt::format("{}", fmt::join(items, "\n"));
+}
+
+int build(const build_args& args,
+          [[maybe_unused]] const global_settings& settings) {
+
+    std::filesystem::path recipe_path{args.uenv_recipe_path};
+
+    if (!std::filesystem::is_directory(recipe_path)) {
+        term::error("{} not a directory", recipe_path);
+        return 1;
+    }
+
+    // check the directory contains the expected files
+    for (std::string f :
+         {"compilers.yaml", "config.yaml", "environments.yaml"}) {
+        if (!std::filesystem::is_regular_file(recipe_path / f)) {
+            term::error("{} doesn't contain {}", recipe_path, f);
+            return 1;
+        }
+    }
+
+    // get system from args or autodetect
+    auto system = site::get_system_name(args.system);
+    if (!system) {
+        term::error("Couldn't auto-detect system name. Please specify "
+                    "`--system` flag.");
+        return 1;
+    }
+
+    auto recipe_tar_path = util::make_temp_dir() / "recipe.tar.gz";
+    auto proc = util::run({"env", "--chdir", recipe_path.string(), "tar",
+                           "--dereference", "-czf", recipe_tar_path, "."});
+    if (proc->rvalue() > 0) {
+        term::error("{}", proc->err.string());
+        return 1;
+    }
+
+    // upload recipe and submit uenv build job
+    auto cicd_endpoint = fmt::format(
+        "https://cicd-ext-mw.cscs.ch/ci/uenv/"
+        "build?system={system}&name={name}&version={version}&uarch={"
+        "uarch}&SPACK_DEVELOP={spack_develop}",
+        fmt::arg("dir", recipe_path.string()), fmt::arg("system", *system),
+        fmt::arg("uarch", args.uarch), fmt::arg("name", args.name),
+        fmt::arg("version", args.version),
+        fmt::arg("spack_develop", args.spack_develop ? "-d" : ""));
+    auto res = util::curl::upload(cicd_endpoint, recipe_tar_path);
+    if (!res) {
+        term::error("uenv build submission failed: {} ", res.error().message);
+        return 1;
+    }
+
+    // print summary returned by CICD-ext
+    term::msg("{}", format_reply(res.value()));
```

Comment:
`nlohmann::json` throws exceptions on when the `.at()` method doesn't find the requested key.
Because the `format_reply` method assumes that the returned result is correctly formatted, how about catching any errors and printing a message that helps debug the issue.

```
try {
    term::msg("{}", format_reply(res.value()));
}
catch (basic_json::out_of_range& e) {
    term::error("unable to parse build service reply - please forward this error message along with the arguments to the CSCS Service Desk.\n{}\n{}",
                      e.what(), res.value());
}
```